spec/ics-4-channel-and-packet-semantics/README.md

```diff
@@ -0,0 +1,757 @@
+  consensusState = get("clients/{connection.clientIdentifier}/consensusState")
+  assert(verifyMembership(
+    consensusState.getRoot(),
+    proofAck,
+    "connections/{connection.counterpartyConnectionIdentifier}/channels/{channel.counterpartyChannelIdentifier}",
+    Channel{OPEN, channel.counterpartyModuleIdentifier, channel.moduleIdentifier,
+            channelIdentifier, timeoutHeight}
+  ))
+  channel.state = OPEN
+  channel.nextTimeoutHeight = 0
+  set(channelKey(connectionIdentifier, channelIdentifier), channel)
+}
+```
+
+The `chanOpenTimeout` function is called by either the handshake-originating
+module or the handshake-confirming module to prove that a timeout has occurred and reset the state.
+
+```typescript
+interface ChanOpenTimeout {
+  connectionIdentifier: Identifier
+  channelIdentifier: Identifier
+  timeoutHeight: uint64
+  proofTimeout: CommitmentProof
+}
+```
+
+```typescript
+function chanOpenTimeout(
+  connectionIdentifier: Identifier, channelIdentifier: Identifier,
+  timeoutHeight: uint64, proofTimeout: CommitmentProof) {
+  channel = get(channelKey(connectionIdentifier, channelIdentifier))
+  connection = get("connections/{connectionIdentifier}")
+  assert(connection.state === OPEN)
+  consensusState = get("clients/{connection.clientIdentifier}/consensusState")
+  assert(consensusState.height >= connection.nextTimeoutHeight)
+  switch channel.state {
+    case INIT:
+      assert(verifyNonMembership(
+        consensusState, proofTimeout,
+        channelKey(connection.counterpartyIdentifier, channel.counterpartyIdentifier)
+      ))
+    case OPENTRY:
+      assert(
+        verifyNonMembership(
+          consensusState, proofTimeout,
+          channelKey(connection.counterpartyIdentifier, channel.counterpartyIdentifier)
+        )
+        ||
+        verifyMembership(
+          consensusState, proofTimeout,
+          channelKey(connection.counterpartyIdentifier, channel.counterpartyIdentifier),
+          Channel{INIT, channel.counterpartyModuleIdentifier, channel.moduleIdentifier,
+                  channelIdentifier, timeoutHeight}
+        )
+      )
+    case OPEN:
+      expected = Channel{OPENTRY, channel.counterpartyModuleIdentifier, channel.moduleIdentifier,
+                         channelIdentifier, timeoutHeight}
+      assert(verifyMembership(
+        consensusState, proofTimeout,
+        channelKey(connection.counterpartyIdentifier, channel.counterpartyIdentifier),
+        expected
+      ))
+  }
+  delete(channelKey(connectionIdentifier, channelIdentifier))
+}
+```
+
+##### Closing handshake
+
+The `chanCloseInit` function is called by either module to initiate
+the channel-closing handshake.
+
+```typescript
+interface ChanCloseInit {
+  connectionIdentifier: Identifier
+  channelIdentifier: Identifier
+  nextTimeoutHeight: uint64
+}
+```
+
+```typescript
+function chanCloseInit(connectionIdentifier: Identifier, channelIdentifier: Identifier, nextTimeoutHeight: uint64) {
```

Comment:
Nit: Looks like functions above are formatted differently.  Identifiers are separated from other params via a new line. Not important but consistently formatting these functions increases readability IMHO.

Reply:
Aye, fixed.